### PR TITLE
Geometry

### LINKED ISSUE #4: Generalize Field Finder Util
**Summary**: Currently the field finder util is geared more specifically towards the 2024 field and related poses. It would be best to generalize this to be used across field designs from year-to-year

### PR DESCRIPTION
**Pull request by linglejack06**
- add fieldfinder utils
- including Is point in triangle
- and is point on triangle (either in or on surrounding lines)

resolves #4

### CODE REVIEW

File: geometry/build.gradle
Comment (jkleiber) [inline geometry/build.gradle:44]: can we add this back so formatting happens?

File: geometry/src/main/java/coppercore/FieldFinder.java
Comment (jkleiber) [inline geometry/src/main/java/coppercore/FieldFinder.java:24]: I feel like this function should be a part of a general `containsPoint` function that any `ConvexPolygon` (an abstract superclass name I just came up with) should be able to implement

File: geometry/src/main/java/coppercore/FieldFinder.java
Comment (jkleiber) [inline geometry/src/main/java/coppercore/FieldFinder.java:35]: See above comment on making point/convex polygon intersections a general function rather than specific to FieldFinder. This would be similar to what you've already done with lines

This will allow FieldFinder to extend beyond just triangles very easily

File: geometry/src/main/java/coppercore/Line.java
Comment (jkleiber) [inline geometry/src/main/java/coppercore/Line.java:15]: What if x2==x1?

(The answer is probably to use linear interpolation and parameterize the line rather than using slope intercept form)

File: math/src/main/java/coppercore/Deadband.java
Comment (jkleiber) [inline math/src/main/java/coppercore/Deadband.java:1]: this file shouldn't be in this PR 👀

**Review by jkleiber — COMMENTED** [review]
first pass review

File: geometry/build.gradle
Comment (linglejack06) [reply geometry/build.gradle:44]: its on line 68

File: geometry/src/main/java/coppercore/FieldFinder.java
Comment (linglejack06) [reply geometry/src/main/java/coppercore/FieldFinder.java:24]: added

File: geometry/src/main/java/coppercore/FieldFinder.java
Comment (linglejack06) [reply geometry/src/main/java/coppercore/FieldFinder.java:35]: done

File: math/src/main/java/coppercore/Deadband.java
Comment (linglejack06) [reply math/src/main/java/coppercore/Deadband.java:1]: oops

File: geometry/src/main/java/coppercore/Line.java
Comment (linglejack06) [reply geometry/src/main/java/coppercore/Line.java:15]: how would i go about using linear interpolation?

Comment (jkleiber) [reply geometry/src/main/java/coppercore/Line.java:15]: So I found a better answer actually:

For 3 points (A, B, and C) if you have distance(A,B) = distance (A, C) + distance(C, B), then C is on the line between A and B

So you just need to do the distance formula a few times, which honestly should be a function of a `Line` anyway.

File: geometry/src/main/java/coppercore/Line.java
Comment (aidnem) [inline geometry/src/main/java/coppercore/Line.java:31]: Say we have a line that goes from (0,1) to (2,3)
The slope of that line would be 1.
However, with this function, if we test whether the point (0.5, 1.5) is on the line, the condition:
`point.getY() == point.getX() * slope` will be false because: 1.5 != 0.5 * 1, even though the point is on the line.
Could we reimplement this function to work differently? What @jkleiber wrote above should work.

File: geometry/src/main/java/coppercore/FieldFinder.java
Comment (aidnem) [inline geometry/src/main/java/coppercore/FieldFinder.java:17]: Do we want to make a FieldFinder be able to contain more than one polygon or do you think that would be unnecessary?

**Review by aidnem — CHANGES_REQUESTED** [review]
Aside from the broken isPointOnLine function, this looks good :)
Once we fix that, request me to review and I'll look over it again and probably approve.

**Review by aidnem — APPROVED** [review]
Looks good to me

**Comment by jkleiber** [comment]
Can we delete the slope variable before we merge this (it is unused and still is prone to divide by 0)

**Review by jkleiber — APPROVED** [review]

**Comment by jkleiber** [comment]
@linglejack06 please merge this when able

**Pull request merged**
